fileupload.go

```diff
@@ -59,7 +67,19 @@ func (f *FileUploadParams) AppendDetails(body io.ReadWriter) (string, error) {
 		}
 	}
 
-	if f.File != nil {
+	// Prefer FileData and Filename
+	if f.FileData != nil && f.Filename != "" {
+		part, err := writer.CreateFormFile("file", filepath.Base(f.Filename))
+		if err != nil {
+			return "", err
+		}
+
+		_, err = io.Copy(part, f.FileData)
+		if err != nil {
+			return "", err
+		}
+		// But allow old value of File field for now.
```

Comment:
I tend to prefer to document my conditions like this too, but gofmt doesn't play nicely with them. Would you mind moving this comment to be combined in with line 70 just because it looks pretty awkward indented into this conditional. Could read something like:

> Support both FileData/Filename and File with the former being the newer preferred version